js/client-teambuilder.js

```diff
@@ -1699,19 +1699,25 @@
 
 			if (template.speciesid === 'meowstic') {
 				smogdexid = 'meowstic-m';
-			} else if (smogdexid === 'rotom' || smogdexid === 'deoxys' || smogdexid === 'kyurem' || smogdexid === 'giratina' || smogdexid === 'shaymin' || smogdexid === 'tornadus' || smogdexid === 'thundurus' || smogdexid === 'landorus' || smogdexid === 'pumpkaboo' || smogdexid === 'gourgeist' || smogdexid === 'arceus' || smogdexid === 'meowstic' || smogdexid === 'hoopa') {
-				if (template.forme) smogdexid += '-' + toId(template.forme);
+			} else if (template.forme) {
+				if (template.speciesid === 'vivillon-fancy' || template.speciesid === 'vivillon-pokeball') smogdexid = 'vivillon';
+				if (template.speciesid === 'keldeo-resolute') smogdexid = 'keldeo';
+				if (template.speciesid === 'basculin-blue-striped') smogdexid = 'basculin';
+				if (template.speciesid === 'greninja-ash') smogdexid = 'greninja';
+				if (template.speciesid === 'pikachu-belle' || template.speciesid === 'pikachu-phd' || template.speciesid === 'pikachu-pop-star' || template.speciesid === 'pikachu-rock-star' || template.speciesid === 'pikachu-libre' || template.speciesid === 'pikachu-cosplay') smogdexid = 'pikachu';
+				if (template.speciesid === 'castform-sunny' || template.speciesid === 'castform-snowy' || template.speciesid === 'castform-rainy') smogdexid = 'castform';
+				smogdexid += '-' + toId(template.forme);
```

Comment:
The reason I just say "um" is because it's something that's usually pretty clear when rereading your code. In general, I think that rereading your code and having the epiphany yourself will make you better at understanding code and programming in the future, instead of having answers spoon-fed at you. In fact, some people feel kind of condescended to if I explain all the things they did wrong in too much detail.

However, if you're still stuck, feel free to ask for a hint or the answer, and I'll be very willing to explain it.

Anyway, this time around, @Slayer95 covered what the problem was.